info/src/main/java/edu/artic/info/AccessMemberCardViewModel.kt

```diff
@@ -0,0 +1,225 @@
+package edu.artic.info
+
+import com.fuzz.rx.bindTo
+import com.fuzz.rx.disposedBy
+import edu.artic.analytics.AnalyticsAction
+import edu.artic.analytics.AnalyticsTracker
+import edu.artic.analytics.EventCategoryName
+import edu.artic.info.AccessMemberCardViewModel.DisplayMode
+import edu.artic.viewmodel.BaseViewModel
+import io.reactivex.Observable
+import io.reactivex.rxkotlin.Observables
+import io.reactivex.rxkotlin.subscribeBy
+import io.reactivex.rxkotlin.withLatestFrom
+import io.reactivex.subjects.BehaviorSubject
+import io.reactivex.subjects.Subject
+import java.net.UnknownHostException
+import javax.inject.Inject
+
+/**
+ * Validates the user provided member information.
+ * Switches [AccessMemberCardFragment] view state based on [DisplayMode].
+ *
+ * When [DisplayMode]
+ * * is [DisplayMode.DisplayForm] form is displayed
+ * * is [DisplayMode.UpdateForm] the filled form is displayed
+ * * is [DisplayMode.DisplayAccessCard] the form is hidden and member information is presented with
+ * barcode.
+ *
+ *
+ * @author Sameer Dhakal (Fuzz)
+ */
+class AccessMemberCardViewModel @Inject constructor(
+        private val service: RetrofitMemberDataProvider,
+        private val infoPreferencesManager: MemberInfoPreferencesManager,
+        private val analyticsTracker: AnalyticsTracker
+) : BaseViewModel() {
+
+    sealed class DisplayMode {
+        object DisplayForm : DisplayMode()
+        class DisplayAccessCard(val memberID: String) : DisplayMode()
+        class UpdateForm(val memberID: String, val zipCode: String) : DisplayMode()
+    }
+
+    /**
+     * Emits events for various loading states.
+     */
+    var loadStatus: BehaviorSubject<LoadStatus> = BehaviorSubject.createDefault<LoadStatus>(LoadStatus.None())
+
+    /**
+     * Form field events.
+     */
+    var zipCode = BehaviorSubject.createDefault<String>("")
+    var memberID = BehaviorSubject.createDefault<String>("")
+    var isValid = BehaviorSubject.createDefault<Boolean>(false)
+
+
+    var displayMode: Subject<DisplayMode> = BehaviorSubject.create()
+
+
+    private var members: Subject<List<MemberInfo>> = BehaviorSubject.create()
+    private var selectedMember: Subject<MemberInfo> = BehaviorSubject.create()
+
+    var expiration: Subject<String> = BehaviorSubject.create()
+    var selectedCardHolder: Subject<String> = BehaviorSubject.create()
+    var membership: Subject<String> = BehaviorSubject.create()
+
+    init {
+
+        /**
+         * Validate the form and enable the sign in button.
+         */
+        Observables.combineLatest(
+                zipCode,
+                memberID)
+                .map { (zipCode, memberID) ->
+                    zipCode.isNotEmpty() && memberID.isNotEmpty()
```

Comment:
...not checking zip code length?

Reply:
No zip code validation requirement is mentioned in the ticket. Neither does the iOS app validates it.

Reply:
Odd. That means this is fine as is.